src/core/public/apm_system.ts

```diff
@@ -87,6 +74,45 @@ export class ApmSystem {
     });
   }
 
+  /* Hold the page load transaction open, until all resources actually finish loading */
+  private holdPageLoadTransaction(apm: ApmBase) {
+    this.pageLoadTransaction = apm.getCurrentTransaction();
+
+    // Keep the page load transaction open until all resources finished loading
+    if (this.pageLoadTransaction && this.pageLoadTransaction.type === 'page-load') {
+      // @ts-expect-error 2339
+      this.pageLoadTransaction.block(true);
+      this.pageLoadTransaction.mark('apmSetup');
+    }
+  }
+
+  /* Returns the number of resources that were loaded from network (vs. from disk cache) */
+  private getLoadedResourcesCount() {
+    try {
+      return window.performance
+        .getEntriesByType('resource')
+        .filter((r: Record<string, any>) => r.initiatorType === 'script' && r.transferSize > 0)
+        .length;
+    } catch (e) {
+      return -1;
+    }
+  }
+
+  /* Close and clear the page load transaction */
+  private closePageLoadTransaction() {
+    if (this.pageLoadTransaction && this.pageLoadTransaction.type === 'page-load') {
+      this.pageLoadTransaction.addLabels({ loadedResources: this.getLoadedResourcesCount() });
+      this.pageLoadTransaction.end();
+      this.pageLoadTransaction = undefined;
+    }
+  }
+
+  private markPageLoadStart() {
```

Comment:
Maybe it makes sense to move all the methods in a separate file to isolate the logic? For example, I had to open the file source code to make sure `this.pageLoadTransaction` is not used outside of the new methods introduced in the current PR.

Reply:
The reason why I didn't want to move this out is becaue the `pageLoad` transaction is implicitly created when calling `init` on line 52. I think it's strange anyway and I think that moving this to a seaparate file makes it too vague.

I added a comment to make this point even more visible.